src/PhpDoc/TypeNodeResolver.php

```diff
@@ -833,6 +852,32 @@ private function resolveConstTypeNode(ConstTypeNode $typeNode, NameScope $nameSc
 		return new ErrorType();
 	}
 
+	private function generateIntMaskType(Type $type): ?Type
+	{
+		$ints = TypeUtils::getConstantIntegers($type);
+		if (count($ints) === 0) {
+			return null;
+		}
+
+		$values = [];
+
+		foreach ($ints as $int) {
+			$int = $int->getValue();
+
+			if ($int !== 0) {
+				foreach ($values as $value) {
+					$values[] = $value | $int;
```

Comment:
One idea for improvement. If the generated integer list is continuous, we could return IntegerRangeType instead.